src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -1766,6 +1766,93 @@ public void testS3PutInGS() throws Throwable {
     }
   }
 
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testGcpAzureReEncryption() throws Throwable {
+    Properties paramProperties = new Properties();
+    List<String> supportedAccounts = Arrays.asList("gcpaccount", "azureaccount");
+    for (String account : supportedAccounts){
+      try (Connection connection = getConnection(account, paramProperties);
+           Statement statement = connection.createStatement()) {
+        try {
+          // create a stage to put the file in
+          statement.execute("CREATE OR REPLACE STAGE " + testStageName);
+
+          // put compressed and uncompressed file
+          String putCompressedCall =
+                  String.format("PUT file://%s @%s", getFullPathFileInResource(TEST_DATA_FILE), testStageName);
+          String putUncompressedCall =
+                  String.format("PUT file://%s @%s auto_compress=false", getFullPathFileInResource(TEST_DATA_FILE), testStageName);
+          statement.execute(putCompressedCall);
+          statement.execute(putUncompressedCall);
+
+          // get unencrypted contents
+          String getCompressedCall =
+                  String.format(
+                          "select system$it('GET_FILE_FROM_STAGE', '%s', '', '%s', '%s')",
```

Comment:
What does this function do? How is it different comparing to regular GET?

Reply:
This is a system function that returns file contents as a row. I wanted to use this to verify that the encrypted contents are different form the non-encrypted ones. JDBC automatically decrypts it so you cannot verify by using normal GET. I will add some more comments here.